src/core/execution/alliance/AllianceExtensionExecution.ts

```diff
@@ -36,27 +45,33 @@ export class AllianceExtensionExecution implements Execution {
       return;
     }
 
-    // Mark this player's intent to extend
-    alliance.addExtensionRequest(this.from);
+    let extend: boolean = false;
 
-    if (alliance.canExtend()) {
```

Comment:
Hey, I noticed that these changes were not being hit by the tests: https://github.com/openfrontio/OpenFrontIO/pull/1582

It might also be good to add a test case to `tests/AllianceExtensionExecution.test.ts` where the recipient is a non-human. 

---

I think that this is the only place that `canExtend` is used. Maybe instead of removing it, an alternative solution could be like this:

```
canExtend(): boolean {
    return (
      this.extensionRequestedRequestor_ && (this.extensionRequestedRecipient_ || this.recipient_.type() !== PlayerType.Human)
    );
  }
```

Thoughts?

Reply:
Ah I see - thanks for the explanation. 

I took a look at your new changes and they look good. Hopefully you caught the last bug on that code path 😆